src/python/pants/backend/python/rules/pytest_coverage.py

```diff
@@ -127,10 +94,12 @@ class PytestCoverageDataCollection(CoverageDataCollection):
     element_type = PytestCoverageData
 
 
+# TODO(#7490): Once we have support for `await Get` of singletons, we can remove this class. We only
```

Comment:
CoverageConfig isn't a singleton, because it's dependent on `TT`. So without this class, it looks like you could `await Get(CoverageConfig, TransitiveTargets, ..)` in this case, which might send a better signal that the config is target-dependent?

Needing to get the `CoveragePlugin` on the other hand would be a good example of a singleton, since it doesn't have any dependencies other than an intrinsic.

Reply:
Yes, but we want the TT to be calculated from the entire target roots closure. It should not be parametrized at the call site. So, it would be wrong to give as an explicit Param in the Get. 

Agreed on CoveragePlugin. It’s so cheap to compute that I decided to have it always be a param. But a good followup would be to make it conditional once we have the new mechanism.

Reply:
> Yes, but we want the TT to be calculated from the entire target roots closure. It should not be parametrized at the call site. So, it would be wrong to give as an explicit Param in the Get.

It would be no more wrong than assuming "all instances of `TransitiveTargets` are always the entire closure"... they aren't, necessarily. Because someone could always project a new instance of that type, which would change its meaning in a subgraph.